src/components/common/Actions.vue

```diff
@@ -241,11 +261,22 @@ export default class Actions extends Mixins(DateMixin, FilingTemplateMixin, Lega
 
       let filingComplete: any
       try {
-        const filing = this.isIncorporationFiling
-          ? await this.buildIncorporationFiling()
-          : await this.buildDissolutionFiling()
+        let filing
+        switch (this.getFilingType) {
+          case FilingTypes.INCORPORATION_APPLICATION:
+            filing = await this.buildIncorporationFiling()
+            break
+          case FilingTypes.REGISTRATION:
+            filing = await this.buildRegistrationFiling()
+            break
+          case FilingTypes.DISSOLUTION:
+            filing = await this.buildDissolutionFiling()
+            break
+        }
 
+        // Save filing
```

Comment:
I see a lot of code duplication now. Can it be simplified easily?

Reply:
Updated in next commit 👍